transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannel.java

```diff
@@ -33,18 +35,18 @@
     private volatile DomainSocketAddress remote;
 
     public EpollDomainSocketChannel() {
-        super(Native.socketDomainFd());
+        super(newSocketDomain());
     }
 
-    public EpollDomainSocketChannel(Channel parent, FileDescriptor fd) {
+    public EpollDomainSocketChannel(Channel parent, Socket fd) {
         super(parent, fd.intValue());
     }
 
     /**
      * Creates a new {@link EpollDomainSocketChannel} from an existing {@link FileDescriptor}
      */
-    public EpollDomainSocketChannel(FileDescriptor fd) {
-        super(fd);
+    public EpollDomainSocketChannel(Socket fd, boolean active) {
```

Comment:
@normanmaurer - I spent a bunch of time debugging an issue related to the socket was marked as active because `getSoError()` returned `0`, but then SSL engine initiated the handshake, which involved doing a write before the channel was connection. This results in a rather nebulous channel closed exception which was difficult to track down (even after figuring out it was a 107 as a result of writeAddress).  I changed the API here just to discuss ... 1. do you have an example of a time when someone creates an FD and it is already active (as determined by getSoError)? 2. If so I will add another constructor which sets active based upon getSoError.